hearts/api/rooms.py

```diff
@@ -0,0 +1,33 @@
+from bson.objectid import ObjectId
+
+from hearts import mongo
+
+
+class RoomDoesNotExist(Exception):
+    pass
+
+
+class RoomCreateFailed(Exception):
+    pass
+
+
+def get_room(room_id):
+    try:
+        if not isinstance(room_id, ObjectId):
+            room_id = ObjectId(room_id)
+    except TypeError:
+        raise TypeError("room_id must be a string or ObjectId, was {}".format(type(room_id).__name__))
+
+    result = mongo.db.rooms.find_one({'_id': room_id})
+    if not result:
+        raise RoomDoesNotExist()
+
+    return result
+
+
+def create_room(users=tuple()):
```

Comment:
As an aside, I think that when the value is written to the database it's later read as a list (at least, the tests which append to the list later on still pass).

However, more directly, you never want to define a function with mutable default arguments, because they can change:

https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

We could use a sentinel if you like, but any immutable value works.